pkg/operations/remove.go

```diff
@@ -32,16 +32,13 @@ func CleanupVM(vm *api.VM) error {
 	// Inspect the container before trying to stop it and it gets auto-removed
 	inspectResult, _ := providers.Runtime.InspectContainer(util.NewPrefixer().Prefix(vm.GetUID()))
 
-	// If the VM is running, try to kill it first so we don't leave dangling containers
-	if vm.Running() {
-		if err := StopVM(vm, true, true); err != nil {
+	// If the VM is running, try to kill it first so we don't leave dangling containers. Otherwise, try to cleanup VM networking.
+	if err := StopVM(vm, true, true, vm.Running()); err != nil {
+		if vm.Running() {
 			return err
 		}
-	} else {
-		// Try to cleanup VM networking
-		if err := removeNetworking(util.NewPrefixer().Prefix(vm.GetUID()), false); err != nil {
-			log.Warnf("Failed to cleanup networking for stopped container %s %q: %v", vm.GetKind(), vm.GetUID(), err)
-		}
+
+		log.Warnf("Failed to cleanup networking for stopped container %s %q: %v", vm.GetKind(), vm.GetUID(), err)
```

Comment:
This warning can be misleading if `StopVM` succeeds in removing network but fails to kill/stop the container.